apps/webapp/src/state/receive.ts

```diff
@@ -0,0 +1,43 @@
+import { AllSlices, SliceCreator } from '.';
+
+export interface ReceiveSlice {
+  index: number;
+  ephemeral: boolean;
+  previous: () => void;
+  next: () => void;
+  setIndex: (index: number) => void;
+  setEphemeral: (ephemeral: boolean) => void;
+}
+
+export const createReceiveSlice = (): SliceCreator<ReceiveSlice> => (set, get) => {
+  return {
+    index: 0,
+    ephemeral: false,
+    previous: () => {
+      const current = get().receive.index;
+      set(state => {
+        if (current > 0) {
+          state.receive.index = current - 1;
+        }
+      });
```

Comment:
Shouldn't the if clause be on the outside?

```typescript
      if (current > 0) {
        set(state => {
          state.receive.index = current - 1;
        });
      }
```

Reply:
yes, but i reuse it from there - https://github.com/penumbra-zone/web/blob/main/apps/extension/src/state/accounts.ts#L24